src/backend.rs

```diff
@@ -163,6 +174,8 @@ pub struct BackendHandler<P> {
     /// The block to fetch data from.
     // This is an `Option` so that we can have less code churn in the functions below
     block_id: Option<BlockId>,
+    /// The mode for fetching account data
+    account_fetch_mode: GetAccountMode,
```

Comment:
I think we can do a little trick here and use an `Arc<AtomicU8>` for this if we represent the various states as u8 values,

e.g. 0 is unchecked, 1 is supports eth_getAccInfo, 2 does not support

then we actually don't even need the enum really and can instead instroduce const u8 for the variants